rust/lance/src/dataset/fragment.rs

```diff
@@ -504,10 +504,13 @@ impl FileFragment {
 
         // scan with predicate and row ids
         let mut scanner = self.scan();
-        scanner
-            .with_row_id()
-            .filter(predicate)?
-            .project::<&str>(&[])?;
+        scanner.with_row_id();
+
+        // if predicate is true, delete the whole fragment
+        // else, filter the predicate
+        if predicate != "true" {
+            scanner.filter(predicate)?.project::<&str>(&[])?;
```

Comment:
Technically, we don't even need to scan. We know we are removing all rows, so we will drop the fragment. We can just early return with `Ok(None)` here.

```suggestion
            return Ok(None);
```